tests/common/debug.py

```diff
@@ -85,3 +88,31 @@ def condition(x):
         settings=settings,
         random=random,
     )
+
+
+def assert_no_examples(strategy, condition=None):
+    @Settings(perform_health_check=False, max_examples=1, max_iterations=100)
+    @given(strategy)
+    def test(s):
+        if condition is not None:
+            assume(condition(s))
+
+    with pytest.raises(Unsatisfiable):
+        test()
+
+
+def assert_all_examples(strategy, predicate):
+    """Checks that there are no examples with given strategy that do not match
```

Comment:
Doesn't quite flow - I'd use something like "Check that ``strategy`` cannot generate an example for which ``predicate`` is False."

Reply:
I think you wrote this code. ;-) I just moved it out of where it was. I'll fix it though.